src/app/guards/hooks.ts

```diff
@@ -55,13 +58,16 @@ const useDirectiveGuard = (
         serverError,
     ]);
 
-    logRocketConsole(
-        `useDirectiveGuard:directiveState:${selectedDirective}`,
-        directiveState
-    );
-
-    const [freshDirective, setFreshDirective] =
-        useState<AppliedDirective<UserClaims> | null>(null);
+    // The memo above was getting called more often than planned and I think it might have
+    //  been part of https://github.com/estuary/ui/issues/999. So the thinking here is that
+    //  we only really care to update if the value actually changed and that might prevent
+    //  extra hook calls that end up getting "confused"
+    useEffect(() => {
+        logRocketEvent(CustomEvents.DIRECTIVE_GUARD_STATE, {
+            state: directiveState,
+        });
```

Comment:
Making this logging an event now so we can easily find sessions where they get a `null` state.